content/algorithms/post/_index.md

```diff
@@ -8,17 +8,117 @@ dashboardAudit: wip
+
+### Constants & Terminology
+
+- **partition:** a group of 2350 sectors proven simultaneously.
+- **proving period:** average period for proving all sectors (currently set to ~24 hours).
+- **deadline:** one of multiple points during a proving period when the proofs for some partitions are due.
+- **challenge window:** the period immediately before a deadline during which a challenge can be generated by the chain and the requisite proofs computed.
+- **miner size:** the amount of proven storage maintained by a single miner actor.
+
+
+Every miner must demonstrate availability of all claimed sectors on a 24hr basis. Constraints on individual proof computations limit a single proof to 2350 sectors (a partition), with 10 challenges each.
+
+As an illustrative example of the current sector size, with a 32 GiB sector, 1 EiB of storage power requires:
+- ~33 million sectors
+- ~14 thousand partitions
+- ~14 thousand proofs per period (day)
+
+### Design
+
+Each miner actor is allocated a 24-hr proving period at random upon creation. This proving period is divided into 48 non-overlapping half-hour deadlines. Each sector is assigned to one of these deadlines at the beginning of the first proving period when the miner registers the sector and never changes deadline. The sets of sectors due at each deadline is recorded in a collection of 48 bitfields.
+
+At the beginning of each period, a cron invocation removes expired sectors and allocates any newly-proven sectors sequentially to one of the 48 deadlines. Sectors are first allocated to fill any deadline up to the next whole-partition multiple (2350) of sectors; next a new partition is started on the deadline with the fewest partitions. If all deadlines have the same number of sectors, a new partition is opened on a random deadline.
+
+The per-deadline sector sets are frozen at the beginning of each proving period as proving set bitfields. The sector IDs are then (logically) divided sequentially into partitions, and the partitions across all deadlines for the miner logically numbered sequentially. Thus, a sector may move between partitions at the same deadline as other sectors fault or expire, but never changes deadline.
+
+If a miner adds 48 partitions worth of sectors (~3.8 PiB) they will have one proof due for each deadline. When a miner has more than 48 partitions, some deadlines will have multiple proofs due at the same deadline. These simultaneous proofs are expected to be computed and submitted together in a single message, at least up to 10-20 partitions per message, but can be split arbitrarily between messages (which, however, will cost more gas).
```

Comment:
> they will have one proof

=> use partition

> some deadlines will have multiple proofs

=> use partitions 

you can explicitly say that a partition is a single SNARK proof.